backend/src/intake-form/intake-form.service.ts

```diff
@@ -20,140 +45,378 @@ export class IntakeFormService {
+      driverLicense: personnel.driverLicense.map((itm) => DriverLicense[itm]),
+      homeLocation: locations?.find(
+        (itm) => itm.id === parseInt(personnel.homeLocation),
+      ),
+      ministry: Ministry[personnel.ministry],
+      division: personnel.division,
+      tools: personnel.tools ? personnel.tools?.map((item) => {
+        
+        const tool = new PersonnelTools();
+        tool.toolId = parseInt(item.toolId);
+        tool.proficiencyLevel = ToolsProficiency[item.toolProficiency];
+        return tool;
+      }) : [],
+      languages: personnel.languages ? personnel.languages.map((item) => {
+        if(item.language === '' && item.languageProficiency === ''){
+          return
+        } 
+        const language = new LanguageEntity();
+        language.language = item.language;
+        language.level = LanguageProficiency[item.languageProficiency];
+        return language;
+      }) : [],
+      certifications: personnel.certifications ? personnel.certifications?.map((item) => {
+        if(item.certificationId === '' && item.expiry === undefined){
+          return
+        }
+        const certification = new PersonnelCertificationEntity();
+        certification.certificationId = parseInt(item.certificationId);
+        certification.expiry = item.expiry ?? undefined;
+        return certification;
+      }) : [],
+      emergencyContactFirstName: personnel.emergencyContactFirstName,
+      emergencyContactLastName: personnel.emergencyContactLastName,
+      emergencyContactPhoneNumber: personnel.emergencyContactPhoneNumber,
+      emergencyContactRelationship: personnel.emergencyContactRelationship,
+    };
+    return Object.assign(personData, person);
+  }
+  /**
+   * Prefill form data with existing data from personnel
+   * @param personnel
+   * @returns
+   */
+  mapPersonnelToForm(personnel: PersonnelEntity): IntakeFormPersonnelData {
+    console.log(personnel, 'OERRS ');
+    return {
+      firstName: personnel.firstName,
+      program:
+        personnel.bcws && personnel.emcr
+          ? Program.ALL
+          : personnel.bcws
+          ? Program.BCWS
+          : Program.EMCR,
+      lastName: personnel.lastName,
+      employeeId: personnel?.employeeId,
+      paylistId: personnel?.paylistId,
+      email: personnel.email,
+      primaryPhone: personnel.primaryPhone,
+      secondaryPhone: personnel.secondaryPhone,
+      workPhone: personnel.workPhone,
+      unionMembership: personnel?.unionMembership,
+      supervisorFirstName: personnel.supervisorFirstName,
+      supervisorLastName: personnel.supervisorLastName,
+      supervisorEmail: personnel?.supervisorEmail,
+      supervisorPhone: personnel?.supervisorPhone,
+      driverLicense: personnel.driverLicense?.map((itm) => itm.toString()),
+      homeLocation: personnel.homeLocation.id.toString(),
+      ministry: personnel.ministry,
+      division: personnel.division,
+      tools: personnel.tools.length === 0 ? [{
```

Comment:
Weird. I'm assuming it's so there's an empty row to start with?

Reply:
yes, it's annoying but I'd rather clean it up here than FE